src/components/Checkbox.tsx

```diff
@@ -1,73 +1,50 @@
-import { MaterialCommunityIcons } from '@expo/vector-icons'
+import { Checkbox as NBCheckbox, ICheckboxProps, FormControl } from 'native-base'
 import { Control, Controller, ControllerRenderProps, FieldErrors, get } from 'react-hook-form'
-import { Pressable, PressableProps } from 'react-native'
 
-import { Row } from './Atoms'
-import { Text } from './Typography'
+import { useCallback } from '~hooks'
 
-import { useCallback, useTheme } from '~hooks'
-
-type CheckboxProps = PressableProps & {
+type CheckboxProps = Omit<ICheckboxProps, 'value'> & {
+  label: string
+  isRequired?: boolean
+  isInvalid?: boolean
   value: boolean
-  onChangeValue: (newValue: boolean) => void
-  label?: string
-  children?: React.ReactNode
-  disabled?: boolean
-  size?: number
-  error?: boolean
-}
-
-interface ControllerFieldProps {
-  field: ControllerRenderProps
 }
 
-export const Checkbox = ({
-  disabled,
-  value,
-  onChangeValue,
+export const Checkbox: React.FC<CheckboxProps> = ({
```

Comment:
I think this is more than a `Checkbox` as it's wrapped with `FormControl`. I imagine that there could be such case when we would like to just use Checkbox component without any wrappers, helper texts, errors. 

Maybe we should split it in such way:
Input, Checkbox, Switch / Toggle, Button, Radio, Select, Slider - raw, but already styled components

Then FormControl components with label, helperText, error. I'd named them Field

I would first create generic Field component and then wrap Checkbox and Input with Field, then I would do:

```ts
Field.Input
Field.Checkbox
Fielx.OtherComponents
```

Last, but not least, we could Create `FieldWithController` to wrap the whole field component with Contoller.

This is just an idea, but IMO, each abstraction should be on it's own, so we could:
- use regular input components without FormControl
- use FormControl without react-hook-form Controller
- use generic Controller components

Let's discuss this further @MSzalowski